bindings/pyfiction/include/pyfiction/inout/write_svg_layout.hpp

```diff
@@ -32,25 +29,35 @@ inline void write_sidb_layout_svg_impl(pybind11::module& m)
                                           const fiction::write_sidb_layout_svg_params&) =
         &fiction::write_sidb_layout_svg<Lyt>;
 
-    // Register the original function to write to a file
+    // SiDB plot
     m.def("write_sidb_layout_svg", write_sidb_layout_svg_pointer, "layout"_a, "filename"_a,
           "ps"_a = fiction::write_sidb_layout_svg_params{}, DOC(fiction_write_sidb_layout_svg));
 
     // Register the function to return an SVG as a string
     m.def(
-        "write_sidb_layout_svg_and_return_as_string",
+        "write_sidb_layout_svg_to_string",
         [](const Lyt& layout, const fiction::write_sidb_layout_svg_params& params) -> std::string
         {
             std::ostringstream oss;                               // Create an in-memory output stream
             fiction::write_sidb_layout_svg(layout, oss, params);  // Write to the stream
             return oss.str();                                     // Return the string content
         },
         "layout"_a, "ps"_a = fiction::write_sidb_layout_svg_params{}, DOC(fiction_write_sidb_layout_svg));
+
+
+    // QCA plot
+    void (*write_qca_layout_svg_pointer)(const py_qca_layout&, const std::string_view&,
+                                              const fiction::write_qca_layout_svg_params&) =
+        &fiction::write_qca_layout_svg<py_qca_layout>;
+
+    m.def("write_qca_layout_svg", write_qca_layout_svg_pointer, "layout"_a, "filename"_a,
+          "params"_a = fiction::write_qca_layout_svg_params{}, DOC(fiction_write_qca_layout_svg));
```

Comment:
Since the QCA SVG writer is defined in this function, it is called and thus registered six times. That could lead to problems. Rather make two `_impl` functions. One for QCA and one for SiDB.